scripts/sct_viewer.py

```diff
@@ -627,38 +631,59 @@ def compute_offset(self):
     def on_press(self, event, plot=None):
         # below is the subplot that refers to the label collection
         if event.inaxes and plot.view == self.orientation[self.primary_subplot]:
-            if self.primary_subplot == 'ax':
-                target_point = Coordinate([int(self.list_slices[self.current_slice]), int(event.ydata) - self.offset[1], int(event.xdata) - self.offset[2], 1])
-            elif self.primary_subplot == 'cor':
-                target_point = Coordinate([int(event.ydata) - self.offset[0], int(self.list_slices[self.current_slice]), int(event.xdata) - self.offset[2], 1])
-            elif self.primary_subplot == 'sag':
-                target_point = Coordinate([int(event.ydata) - self.offset[0], int(event.xdata) - self.offset[1], int(self.list_slices[self.current_slice]), 1])
+            if not self.enable_custom_points:
+                if self.primary_subplot == 'ax':
+                    target_point = Coordinate([int(self.list_slices[self.current_slice]), int(event.ydata) - self.offset[1], int(event.xdata) - self.offset[2], 1])
+                elif self.primary_subplot == 'cor':
+                    target_point = Coordinate([int(event.ydata) - self.offset[0], int(self.list_slices[self.current_slice]), int(event.xdata) - self.offset[2], 1])
+                elif self.primary_subplot == 'sag':
+                    target_point = Coordinate([int(event.ydata) - self.offset[0], int(event.xdata) - self.offset[1], int(self.list_slices[self.current_slice]), 1])
+
+            else:
+                if self.primary_subplot == 'ax':
+                    target_point = Coordinate([int(self.current_point.x), int(event.ydata) - self.offset[1], int(event.xdata) - self.offset[2], 1])
```

Comment:
The CoordinateValue class has a mode called `mode=index`, is it not more expressive using this class instead?